implementing-nosto/advanced-implementation/accessing-state-with-selectors.md

```diff
@@ -0,0 +1,61 @@
+# Accessing state with selectors
+
+In Nosto, accessing the app state is simple. `useAppState` hook provides you full access to the entire application state.
+
+{% code title="State example" %}
+```jsx
+import { useAppState } from '@nosto/preact'
+
+export default () => {
+    const state = useAppState()
+	...
+}
+```
+{% endcode %}
+
+While simple, this approach has a number of disadvantages. Namely, any change to the state will trigger a re-render on every component that uses the state; and it may be awkward to access data that does not reside in a top level property.
+
+{% code title="Example" %}
+```jsx
+const { response: { products }} = useAppState()
+```
+{% endcode %}
+
+## Selector hook
+
+The solution to the issues describe above is the `useAppStateSelector` hook. If you have used selectors from other popular state management libraries, then you are already familiar with the concept.
```

Comment:
We can remove this section. Starting from next section makes more sense

Reply:
Delete the entire selector hook section? Or do you mean paragraph?

Reply:
Only the paragraph :)